### PR TITLE
TST: concat(..., copy=False) with datetime tz-aware data raises ValueError: cannot create a DatetimeTZBlock without a tz

### LINKED ISSUE #25257: concat(..., copy=False) with datetime tz-aware data raises ValueError: cannot create a DatetimeTZBlock without a tz
#### Code Sample, a copy-pastable example if possible

```python
import pandas as pd
df = pd.DataFrame({
    'timestamp': [pd.Timestamp('2017-08-27 01:00:00.709949+0000', tz='UTC')],
})

pd.concat([df])      # works
pd.concat([df, df])  # works
pd.concat([df], copy=False)  # fails
# Edit: added full traceback
# Traceback (most recent call last):
#   File "<stdin>", line 1, in <module>
#   File ".../pandas/core/reshape/concat.py", line 229, in concat
#     return op.get_result()
#   File ".../pandas/core/reshape/concat.py", line 426, in get_result
#     copy=self.copy)
#   File ".../pandas/core/internals/managers.py", line 2055, in concatenate_block_managers
#     b = b.make_block_same_class(values, placement=placement)
#   File ".../pandas/core/internals/blocks.py", line 235, in make_block_same_class
#     klass=self.__class__, dtype=dtype)
#   File ".../pandas/core/internals/blocks.py", line 3095, in make_block
#     return klass(values, ndim=ndim, placement=placement)
#   File ".../pandas/core/internals/blocks.py", line 1680, in __init__
#     values = self._maybe_coerce_values(values)
#   File ".../pandas/core/internals/blocks.py", line 2266, in _maybe_coerce_values
#     raise ValueError("cannot create a DatetimeTZBlock without a tz")
# ValueError: cannot create a DatetimeTZBlock without a tz

```
#### Problem description
Running `pd.concat` with a single dataframe and `copy=False` fails, giving an error about  `cannot create a DatetimeTZBlock without a tz`.

Closely related:
- https://github.com/pandas-dev/pandas/issues/19420
- https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3AReshaping+label%3ATimezones

#### Expected Output

The original dataframe, since I'm only `concat`-ing a single dataframe.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit: None
python: 3.6.8.final.0
python-bits: 64
OS: Linux
OS-release: 4.18.0-14-generic
machine: x86_64
processor: x86_64
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

pandas: 0.24.1
pytest: 4.2.0
pip: 19.0.1
setuptools: 40.6.3
Cython: 0.29.4
numpy: 1.15.4
scipy: 1.2.0
pyarrow: 0.11.1
xarray: None
IPython: 7.2.0
sphinx: None
patsy: None
dateutil: 2.7.5
pytz: 2018.7
blosc: None
bottleneck: None
tables: None
numexpr: None
feather: None
matplotlib: 3.0.2
openpyxl: 2.4.10
xlrd: 1.2.0
xlwt: None
xlsxwriter: 1.1.2
lxml.etree: None
bs4: None
html5lib: None
sqlalchemy: 1.2.17
pymysql: None
psycopg2: 2.7.6.1 (dt dec pq3 ext lo64)
jinja2: 2.10
s3fs: None
fastparquet: 0.2.1
pandas_gbq: None
pandas_datareader: None
gcsfs: None

</details>

### PR DESCRIPTION
**Pull request by SurajH1**
…n version 0.24.1 but passes on current version

- [x] closes #25257
- [x] tests added / passed

### CODE REVIEW

File: pandas/tests/dtypes/test_concat.py [hunk 1]
```diff
@@ -88,3 +88,12 @@ def test_concat_mismatched_categoricals_with_empty():
     result = _concat.concat_compat([ser1._values, ser2._values])
     expected = pd.concat([ser1, ser2])._values
     tm.assert_categorical_equal(result, expected)
+
+
+# This will raise a ValueError issue if it fails
```
Comment (WillAyd) [inline pandas/tests/dtypes/test_concat.py:93]: Can you move this comment to the body of the function?

File: pandas/tests/dtypes/test_concat.py [hunk 2]
```diff
@@ -88,3 +88,12 @@ def test_concat_mismatched_categoricals_with_empty():
...
+# Regression test for issue 25257
+def test_no_tz_concat_without_copy():
+    df = pd.DataFrame({
+        'timestamp': [pd.Timestamp('2020-04-08 09:00:00.709949+0000', tz='UTC')],
+    })
+    pd.concat([df], copy=False)
```
Comment (WillAyd) [inline pandas/tests/dtypes/test_concat.py:99]: Our typical testing strategy is to construct the result and expected dataframes, and finish with a `tm.assert_frame_equal(result, expected)`; you can look at other tests if you need more guidance, but here should do `result = pd.concat([df], copy=False)` and construct `expected` separately before comparing

**Review by WillAyd — CHANGES_REQUESTED** [review]
Thanks - looks good. Some minor feedback

File: pandas/tests/dtypes/test_concat.py (hunk 2 above)
Comment (SurajH1) [reply pandas/tests/dtypes/test_concat.py]: resolved

File: pandas/tests/dtypes/test_concat.py (hunk 1 above)
Comment (SurajH1) [reply pandas/tests/dtypes/test_concat.py]: resolved

File: pandas/tests/dtypes/test_concat.py [hunk 3]
```diff
@@ -88,3 +88,13 @@ def test_concat_mismatched_categoricals_with_empty():
...
     expected = pd.concat([ser1, ser2])._values
     tm.assert_categorical_equal(result, expected)
+
+
+def test_no_tz_concat_without_copy():
+    # This will raise a ValueError issue if it fails
+    # Regression test for issue 25257
+    df = pd.DataFrame({'timestamp': [pd.Timestamp('2020-04-08 09:00:00.709949+0000',
+                      tz='UTC')], })
+    result = pd.concat([df], copy=False)
```
Comment (jreback) [inline pandas/tests/dtypes/test_concat.py:98]: can you parameterize on copy=[True, False]

**Review by jreback — CHANGES_REQUESTED** [review]
pls merge master and ping on green.

File: pandas/tests/dtypes/test_concat.py (hunk 3 above)
Comment (SurajH1) [reply pandas/tests/dtypes/test_concat.py]: Resolved

**Comment by SurajH1** [comment]
I'm not sure why the  Publish benchmarks artifact error is popping up on Travis-CI tests. Any help would be appreciated @jreback

**Comment by jreback** [comment]
can you merge master

File: pandas/core/computation/expr.py [hunk 4]
```diff
@@ -670,19 +670,26 @@ def visit_Call(self, node, side=None, **kwargs):
                 )
 
             return res(*new_args)
+        elif isinstance(res, np.ufunc):
+            new_args = [self.visit(arg) for arg in node.args]
```
Comment (jreback) [inline pandas/core/computation/expr.py:674]: what is this for / from?

File: pandas/tests/computation/test_eval.py [hunk 5]
```diff
@@ -727,6 +727,10 @@ def test_line_continuation(self):
         result = pd.eval(exp, engine=self.engine, parser=self.parser)
         assert result == 12
 
+    def test_floor_expression(self):
```
Comment (jreback) [inline pandas/tests/computation/test_eval.py:730]: what is this?

File: pandas/tests/dtypes/test_concat.py (hunk 3 above)
Comment (jreback) [reply pandas/tests/dtypes/test_concat.py:98]: can you push up the change

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by simonjayhawkins** [comment]
@SurajH1 i've pushed a few changes to the branch of this PR. I notice you raised the PR from master. In future you should create a seperate branch, see https://pandas.pydata.org/docs/dev/development/contributing.html#creating-a-branch

**Review by simonjayhawkins — APPROVED** [review]
Thanks @SurajH1 lgtm. @jreback @WillAyd

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks @SurajH1 and @simonjayhawkins !

**Pull request merged**